sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -116,6 +116,18 @@
 @Experimental(Experimental.Kind.SOURCE_SINK)
 public class ElasticsearchIO {
 
+  private static void checkVersion(ConnectionConfiguration connectionConfiguration)
```

Comment:
Previously `checkVersion` was in `ConnectionConfiguration` class. I cannot move it back to that class because otherwise version check would be done at configuration time and all the point of this PR is to do the check at run time and not at configuration time. But I agree, private method needs to be at the bottom, I'll move the method at the bottom of `ElasticsearchIO` class